File: app/src/main/java/com/jackzhao/simplekv/MyConfig.java
package com.jackzhao.simplekv;

import com.jackzhao.simple_kv.IBaseKv;

public enum MyConfig implements IBaseKv {
    IS_FIRST("is_first", true),
    ;
    String key;
    Object defaultValue;

    MyConfig(String key, Object defValue) {
        this.key = key;
        this.defaultValue = defValue;
    }

}

File: app/src/main/java/com/jackzhao/simplekv/MyConfig.kt
package com.jackzhao.simplekv

import com.jackzhao.simple_kv.IBaseKv
import java.util.HashSet

enum class MyConfig(
    var defaultValue: Any
) : IBaseKv {
    IS_FIRST(true),
    LUNCH_TIMES(0),
    NAME("no name"),
    TMP_HASH(HashSet<String>()),
    ;
}